examples/graph/deckgl-overlay.js

```diff
@@ -34,8 +34,8 @@ export default class DeckGLOverlay extends Component {
 
     // recalculate viewport on container size change.
     const left = -width / 2;
-    const top = -height / 2;
-    const glViewport = new OrthographicViewport({width, height, left, top});
+    const bottom = -height / 2;
+    const glViewport = new OrthographicViewport({width, height, left, bottom});
```

Comment:
For 1. glViewport is a legacy name when we had both SVG viewport and GL viewport, I agree we should change it to just _viewport_.

For 2. the most common use cases we have for the orthographic viewport now is 2D InfoVis, in which the world coordinates == the screen coordinates by default (consider we are just drawing SVG but using WebGL as renderer), and we use a scale factor if users want to zoom in or out (not distance or camera position).

Regarding the props, I'm fine with either
a) {left, top, width, height}, which maps to {x, y, width, height} as the props for SVG rect, or
b) {left, top, bottom, right}, which maps to {x0, y0, x1, y1} as for SVG line
a) is more preferable but the current API support both anyways.

Reply:
when we specify viewport, it is an OpenGL world, up is +y, right is +z, better to align left with bottom, right with top, like {left, bottom, width, height}, {left, right, bottom, top}
Here is glOrtho API from OpenGL lib as reference
glOrtho(GLdouble left, GLdouble right, GLdouble bottom, GLdouble top, GLdouble near, GLdouble far);

Reply:
I get it in the GL world coordinates starts from bottom-left. My point is we should design and expose APIs that fits users' mental model and common practices.

If we were to just follow the low-level GL definitions, fine, we need to clearly document why, given same data as input, the rendering result from Deck.gl and SVG (e.g. d3, react-vis) are not compatible, and the caveat that the users need to do the y-flip themselves if they interleave Deck.gl and SVG layers.

I can take a sweep for our internal applications and hopefully there're not many external ones using the orthographic viewport pre-v4.2. But for accepting this PR, I'd suggest we at least include the following two:
1. highlight the change that we are using _bottom_ instead of _top_ in the v4.1 -> v4.2 upgrade guide;
2. add warning/error message when only _top_ but no _bottom_ prop is specified when instantiating a new orthographic viewport.

Reply:
Thanks, the two suggestions make sense.

Reply:
By the way, the y-flip is a different issue, here we focus on the way how to specify the viewport. The two things are totally different.